src/0install-solver/diagnostics.ml

```diff
@@ -424,17 +427,16 @@ struct
     report
   ;;
 
-  let pp_rolemap ~verbose f reasons =
-    let pp_item f (_, c) = pf f "- @[%a@]" (Component.pp ~verbose) c in
-    Format.(pp_print_list ~pp_sep:pp_print_cut) pp_item f (RoleMap.bindings reasons)
+  let pp_rolemap ~verbose reasons =
+    let pp_item (_, c) = Pp.text "- " ++ Pp.box (Component.pp ~verbose c) in
+    Pp.concat_map ~sep:Pp.cut (RoleMap.bindings reasons) ~f:pp_item
   ;;
 
   (** Return a message explaining why the solve failed. *)
   let get_failure_reason ?(verbose = false) result =
     let+ reasons = of_result result in
-    Format.asprintf
-      "Can't find all required implementations:@\n@[<v0>%a@]"
-      (pp_rolemap ~verbose)
-      reasons
+    Pp.verbatim "Can't find all required implementations:"
+    ++ Pp.newline
```

Comment:
This should be a break with a vbox surrounding the Pp so that it is formatted the correct way. You might need a `hovbox` on the sentence above.